code/game/gamemodes/wizard/artefact.dm

```diff
@@ -567,3 +567,74 @@ var/global/list/multiverse = list()
 	heal_brute = 25
 	heal_burn = 25
 	heal_oxy = 25
+
+//Warp Whistle: Provides uncontrolled long distance teleportation.
+
+/obj/item/warpwhistle
+	name = "warp whistle"
+	desc = "One toot on this whistle will send you to a far away land!"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "whistle"
+	var/on_cooldown = 0 //0: usable, 1: in use, 2: on cooldown
+	var/mob/living/carbon/last_user
+
+/obj/item/warpwhistle/attack_self(mob/living/carbon/user)
+	if(!istype(user) || on_cooldown)
+		return
+	on_cooldown = TRUE
+	last_user = user
+	var/turf/T = get_turf(user)
+	playsound(T,'sound/magic/WarpWhistle.ogg', 200, 1)
+	user.canmove = 0
+	PoolOrNew(/obj/effect/overlay/temp/tornado,T)
+	sleep(20)

```

Comment:
well it does if the src doesn't iirc. But if the object is just qdeleted then it wont.